### PR TITLE
CDAP-1664: Refactor exception hierarchy to use Id.* rather than plain strings

### PR DESCRIPTION
**Pull request by jwang47**
Based on https://github.com/caskdata/cdap/pull/1452 (merged) and https://github.com/caskdata/cdap/pull/1476 (merged)

### CODE REVIEW

**Comment by jwang47** [comment]
Build passed http://builds.cask.co/browse/CDAP-DUT928-6

File: cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java [hunk 1]
```diff
@@ -278,16 +279,29 @@ public void testDataset() throws Exception {
 
   @Test
   public void testProcedure() throws Exception {
-    String qualifiedProcedureId = String.format("%s.%s", FakeApp.NAME, FakeProcedure.NAME);
-    testCommandOutputContains(cli, "start procedure " + qualifiedProcedureId, "Successfully started Procedure");
-    assertProgramStatus(programClient, FakeApp.NAME, ProgramType.PROCEDURE, FakeProcedure.NAME, "RUNNING");
+    String originalApiVersion = cliConfig.getClientConfig().getApiVersion();
     try {
+<<<<<<< HEAD
```
Comment (poornachandra) [inline cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java]: Need to resolve conflict here.

File: cdap-client/src/main/java/co/cask/cdap/client/MetricsClient.java [hunk 2]
```diff
@@ -196,6 +196,7 @@ private long getTotalCounter(Map<String, String> context, String metricName) {
     } catch (Exception e) {
       throw Throwables.propagate(e);
     }
+>>>>>>> 8d3682e1e33f94797a127d225b59a9c29cb7b300
```
Comment (poornachandra) [inline cdap-client/src/main/java/co/cask/cdap/client/MetricsClient.java]: Conflict marker.

**Comment by jwang47** [comment]
Need to fix merge conflicts. Closing for now.

**Pull request closed without merging**